Add Receptor Installation settings select

zkayyali812 opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

This PR adds in 2 new settings - 
1. RECEPTOR_INSTALLATION_METHODS = ['release', 'package', 'local', 'container']
2. DEFAULT_RECEPTOR_INSTALLATION_METHOD = 'release'

`DEFAULT_RECEPTOR_INSTALLATION_METHOD` can be overridden by users to provide an alternate default

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New or Enhanced Feature

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - UI
 - Docs
 - Other

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
➜  awx git:(zk-add-receptor-settings) ✗ make version     
make: *** No rule to make target `version'.  Stop.
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

zkayyali812 committed:
Add Receptor Install settings

zkayyali812 committed:
Add default receptor installation method

zkayyali812 committed:
Allow default receptor install method to be selected

zkayyali812 committed:
Update ha.py

TheRealHaoLiu commented on awx/main/management/commands/host_metric.py:
```diff
@@ -35,15 +35,7 @@ def host_metric_queryset(self, result, offset=0, limit=BATCHED_FETCH_COUNT):
-                'hosts_deleted',
-                'indirectly_managed_hosts',
-            ).order_by(
+            result.values('id', 'date', 'license_consumed', 'license_capacity', 'hosts_added', 'hosts_deleted', 'indirectly_managed_hosts',).order_by(
```
revert this change i think your editor setting is conflicting with black

TheRealHaoLiu commented:
```
/api/v2/instances/                   | instance                   | receptor_installation_method                              |  X  |        | Failed, option mismatch
```
awx-collection completeness check failed

chrismeyersfsu commented on awx/settings/defaults.py:
```diff
@@ -967,6 +967,10 @@
 # K8S only. Use receptor_log_level on AWX spec to set this properly
 RECEPTOR_LOG_LEVEL = 'info'
 
+RECEPTOR_INSTALLATION_METHODS = ['release', 'package', 'local', 'container']
```
I don’t see this variable used. Can we delete it?

TheRealHaoLiu replied:
would be used by the UI to populate the selection drop down 

also somewhere we want to be able to limit what options are "exposed" for example downstream "local" and "release" wont be available

TheRealHaoLiu committed:
Dynamically populate choices for receptor_installation_method

Base on RECEPTOR_INSTALLATION_METHODS and DEFAULT_RECEPTOR_INSTALLATION_METHOD settings

TheRealHaoLiu commented on awx/main/migrations/0189_instance_receptor_installation_method.py:
```diff
@@ -0,0 +1,27 @@
+                    ('local', 'Install from local source'),
+                    ('container', 'Install via container'),
+                ],
+                default='release',
```
```suggestion
                default='package',
```

zkayyali812 committed:
Merge branch 'feature_awx-mesh-ingress' into zk-add-receptor-settings
